frontend/packages/dev-console/src/utils/application-utils.ts

```diff
@@ -384,31 +382,34 @@ export const cleanUpWorkload = (
     models.forEach((model) => deleteRequest(model, resourceObj));
   };
   if (isKnativeResource) {
-    // delete knative resources
-    const knativeService = _.find(workload.resources.ksservices, { kind: 'Service' });
-    resourceData.metadata.name = _.get(knativeService, 'metadata.name', '');
-    batchDeleteRequests(knativeDeleteModels, resourceData);
+    // skip revisions and delete other knative resources
+    // revisions will be deleted automatically when service is deleted
+    if (resource.kind !== KnativeRevisionModel.kind && resource.kind !== KnativeServiceModel.kind) {
+      // crd based resource deletion
+      deleteRequest(modelFor(referenceFor(resource)), resource);
+    }
+    if (resource.kind === KnativeServiceModel.kind) {
+      batchDeleteRequests(knativeDeleteModels, resourceData);
+      webhooksAvailable = true;
+    }
```

Comment:
it'll be better if we can more explicit with models in case of eventing and add under the single switch and have one for service as well

Reply:
refactored the code to be explicit in processing the models and also handled crd/non-crd based deletion in a generic way. 
cc: @sahil143, @invincibleJai